src/promptflow/promptflow/_core/operation_context.py

```diff
@@ -133,10 +120,16 @@ def append_user_agent(self, user_agent: str):
         Args:
             user_agent (str): The user agent information to append.
         """
-        agent = self.get(self._USER_AGENT, '')
-        user_agent = user_agent.strip()
-        if user_agent not in agent:
-            self[self._USER_AGENT] = f"{agent} {user_agent}".strip()
+        if "user_agent" in self:
+            if user_agent not in self.user_agent:
+                self.user_agent = f"{self.user_agent} {user_agent}"
+        else:
+            self.user_agent = user_agent
+        self.user_agent = self.user_agent
```

Comment:
Remove the redundant line at the end of the method. It serves no purpose and could potentially confuse someone reading the code.

Reply:
add a line before  self.user_agent = self.user_agent?

Reply:
done